### ISSUE TITLE
Access response headers

### ISSUE THREAD

**Issue by jamiesoncj** [ngonzalvez/rest-facade#27]
Is it possible to access the response headers as well as the response body in some way?

I would be happy to submit a PR but am not sure the best place to begin approaching this - perhaps you might be able to make a suggested approach?

An example use case (at least for me) for this is when traversing a long list of resources in a `.getAll()` type call. The API I am using sets headers in the response to help with traversing the resources when there are more than can be returned in a single response. For example, a call like `GET /movies` may have a long list of resources returned in the body but only a maximum of 100 resources can be returned in one go. Therefore the response headers `Total-Count: 5000` and `Link` header ([see info](http://tools.ietf.org/html/rfc5988)) are set.

Having access to these response headers would be extremely helpful in order to know how many total resources are available, and to be able to set (for example) `offset` and `limit` parameters in the query string for subsequent requests without too much guesswork.

If you'd accept a PR for this, please let me know - I am not sure what the best approach would be for implementation so any suggestions would be appreciated. Thank you!

**Comment by ngonzalvez** [comment]
Hi @jamiesoncj, PRs are always welcome. However since this was very easy to implement I just did it and released it. Now promises will resolve with the headers as a second argument, and callbacks with headers as a third.

```js
Movies
  .getAll()
  .then((body, headers) => {
     // Some code over here...
  });

// or

Movies.getAll((err, body, headers) => {
     // Some code over here...
});
```

**Comment by ngonzalvez** [comment]
By the way, this is included in version 1.9.0.

**Comment by jamiesoncj** [comment]
Hey @ngonzalvez, that is brilliant - thank you!
